contracts/contracts/strategies/CurvePoolBoosterDirect.sol

```diff
@@ -0,0 +1,259 @@
+
+        // Create the campaign
+        campaignId = votemarket.createCampaign(
+            block.chainid,
+            gauge,
+            address(this),
+            address(rewardToken),
+            _numberOfPeriods,
+            _maxRewardPerVote,
+            balanceSubFee,
+            _blacklist,
+            address(0),
+            true
+        );
+
+        emit CampaignCreated(
+            campaignId,
+            gauge,
+            address(rewardToken),
+            _maxRewardPerVote,
+            balanceSubFee
+        );
+
+        return campaignId;
+    }
+
+    /// @notice Manage the campaign created
+    /// @dev This will use all token available in this contract
+    /// @param _extraNumberOfPeriods Number of additional periods (cannot be 0)
+    ///         that will be added to already existing amount of periods.
+    /// @param _newMaxRewardPerVote New maximum reward per vote
+    /// @param _increaseTotalReward Boolean to increase the total reward amount or not
+    function manageCampaign(
+        uint8 _extraNumberOfPeriods,
+        uint256 _newMaxRewardPerVote,
+        bool _increaseTotalReward
+    ) external nonReentrant onlyGovernorOrStrategist {
+        require(campaignId != 0, "Campaign not created");
+
+        // Handle fee (if any)
+        uint256 balanceSubFee;
+        if (_increaseTotalReward) {
+            balanceSubFee = _handleFee();
+
+            // Approve the reward token to votemarket
+            rewardToken.safeApprove(address(votemarket), 0);
+            rewardToken.safeApprove(address(votemarket), balanceSubFee);
+        }
+
+        // Update the campaign
+        votemarket.manageCampaign(
+            campaignId,
+            _extraNumberOfPeriods,
+            balanceSubFee,
+            _newMaxRewardPerVote
+        );
+
+        emit CampaignManaged(
+            campaignId,
+            _extraNumberOfPeriods,
+            _newMaxRewardPerVote,
+            balanceSubFee
+        );
+    }
+
+    /// @notice Close the campaign.
+    function closeCampaign() external nonReentrant onlyGovernorOrStrategist {
+        require(campaignId != 0, "Campaign not created");
+
+        // Close the campaign
+        votemarket.closeCampaign(campaignId);
+
+        // Reset the campaignId
+        delete campaignId;
+
+        emit CampaignClosed(campaignId);
+    }
+
+    /// @notice calculate the fee amount and transfer it to the feeCollector
+    /// @return Balance after fee
+    function _handleFee() internal returns (uint256) {
+        // Cache current rewardToken balance
+        uint256 balance = rewardToken.balanceOf(address(this));
+        require(balance > 0, "No reward to manage");
+
+        uint256 feeAmount = (balance * fee) / FEE_BASE;
+
+        // If there is a fee, transfer it to the feeCollector
+        if (feeAmount > 0) {
+            // Transfer the fee to the feeCollector
+            rewardToken.safeTransfer(feeCollector, feeAmount);
```

Comment:
⚪ `feeCollector` is being read twice. Could store it in a variable to save some gas. But since this is L2, it's probably okay

Reply:
You are right, but we can fix it: https://github.com/OriginProtocol/origin-dollar/pull/2389/commits/8f600f6ca9a543aeffcc27144c5c1bfa2962ec79.